clang/include/clang/Basic/DiagnosticSemaKinds.td

```diff
@@ -125,6 +125,9 @@ def warn_float_compare_literal : Warning<
 def warn_double_const_requires_fp64 : Warning<
   "double precision constant requires %select{cl_khr_fp64|cl_khr_fp64 and __opencl_c_fp64}0, "
   "casting to single precision">;
+def warn_imp_float_size_conversion : Warning<
+  "implicit conversion between floating point types of different sizes.">,
```

Comment:
```suggestion
  "implicit conversion between floating point types of different sizes">,
```
Diagnostics in clang don't end with trailing punctuation (except for question marks, lol).